crypto/cms/cms_lib.c

```diff
@@ -92,6 +92,8 @@ BIO *CMS_dataInit(CMS_ContentInfo *cms, BIO *icont)
 
     default:
         CMSerr(CMS_F_CMS_DATAINIT, CMS_R_UNSUPPORTED_TYPE);
+        if (!icont)
+            BIO_free(cont);
         return NULL;
```

Comment:
just do
```
goto err;
```
add add label at bottom for the identical code.
```
err:
```

This should also be fixed in master

Reply:
Changed.